--- a/src/compositor/extensions/extensions.pri
+++ b/src/compositor/extensions/extensions.pri
@@ -8,6 +8,7 @@
     ../extensions/touch-extension.xml \
     ../extensions/qt-key-unstable-v1.xml \
     ../extensions/qt-windowmanager.xml \
+     ../extensions/qt-texture-sharing-unstable-v1.xml \
     ../3rdparty/protocol/text-input-unstable-v2.xml \
     ../3rdparty/protocol/viewporter.xml \
     ../3rdparty/protocol/scaler.xml \
@@ -98,8 +99,6 @@
         extensions/qwaylandxdgshellintegration.cpp \
 
     qtConfig(opengl) {
-        WAYLANDSERVERSOURCES += \
-            ../extensions/qt-texture-sharing-unstable-v1.xml
 
         HEADERS += \
             extensions/qwltexturesharingextension_p.h

